aspnetcore/blazor/includes/js-interop/circuit-disconnection.md

```diff
@@ -0,0 +1,46 @@
+*This section only applies to Blazor Server apps.*
+
+JavaScript (JS) interop calls can't be issued after a SignalR circuit is disconnected. Without a circuit during component disposal or at any other time that a circuit doesn't exist, the following method calls fail and log a message that the circuit is disconnected as a <xref:Microsoft.JSInterop.JSDisconnectedException>:
+
+* JS interop method calls
+  * <xref:Microsoft.JSInterop.IJSRuntime.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeAsync%2A?displayProperty=nameWithType>
+  * <xref:Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync%2A?displayProperty=nameWithType>)
+* `Dispose`/`DisposeAsync` calls on any <xref:Microsoft.JSInterop.IJSObjectReference> or `Dispose` calls on any <xref:Microsoft.JSInterop.DotNetObjectReference>.
+
+In order to avoid logging <xref:Microsoft.JSInterop.JSDisconnectedException> or to log custom information, catch the exception in a [`try-catch`](/dotnet/csharp/language-reference/keywords/try-catch) statement.
+
+For the following component disposal example with an <xref:Microsoft.JSInterop.IJSObjectReference> object used to call JS from .NET:
+
+* The component implements <xref:System.IAsyncDisposable>.
+* `objInstance` is an <xref:Microsoft.JSInterop.IJSObjectReference>.
+* <xref:Microsoft.JSInterop.JSDisconnectedException> is caught and not logged.
+* Optionally, you can log custom information in the `catch` statement at whatever log level you prefer. The following example doesn't log custom information because it assumes the developer doesn't care about when or where circuits are disconnected during component disposal.
+
+```csharp
+async ValueTask IAsyncDisposable.DisposeAsync()
+{
+    try
+    {
+        if (objInstance is not null)
+        {
+            await objInstance.DisposeAsync();
+        }
+    }
+    catch (JSDisconnectedException)
+    {
+    }
+}
+```
+
+> [!NOTE]
+> The pattern for <xref:Microsoft.JSInterop.DotNetObjectReference> disposal is the same in scenarios where .NET is called from JS.
+>
+> When a circuit fails and developer disposal code fails with a caught or uncaught <xref:Microsoft.JSInterop.JSDisconnectedException>, automatic garbage collection eventually disposes of any objects created by the developer in the component.
```

Comment:
I'm not sure why we need to point this out here. It's the same as how GC behaves anywhere.

Reply:
Trapped or not, I felt it would be a good idea to tell readers that just because there's a `JSDisconnectedException` when they attempt to dispose (i.e., the dispose fails) that they don't need to sweat that object lingering on. This puts the mind at ease on it.

Reply:
> I'm not sure why we need to point this out here. It's the same as how GC behaves anywhere.

Unless there is a collection of all JSObjectReferences tracked somewhere in blazor internals, which would prevent GC. 

I believe there is a such collection must exists somewhere on the javascript side. But the way, Blazor keeps references to all DotNetObjectReference in .NET memory and they are not automatically disposed by GC, are they?

Reply:
@Liero ... We don't work closed issues/PRs. You can open a new docs issue from the bottom of the topic cross-linking to this PR's updates.

Note that we're on the edge of the .NET 7 release, and the PU can't be pinged on doc issues until a few weeks after the release. Even then, it's not clear if they'll be able to respond with devs taking off for holiday breaks. Such an issue might not be reached until 23Q1.

I'll need to lock this PR down as resolved. Our work process isn't efficient for closed issues/PRs.